manifests/exploded_war.pp

```diff
@@ -37,6 +37,12 @@
     content => template('i2b2/log4j.properties.erb'),
   }
 
+  exec { 'replace-password' :
+    cwd         => "$dir/WEB-INF/conf",
+    command     => 'sed -i \'s#<parameter name="password">axis2</parameter>#<parameter name="password">Z61OaOvXAfsyIJrQ</parameter>#\' axis2.xml',
+    refreshonly => true,
+  }
+
```

Comment:
* You would need to use `subscribe` instead of `require`. But that's not such a great solution. More on that later.
* The other passwords are passwords for database users. The database is not exposed to the network. For better security, the passwords can still be changed via hiera. If it's critical that no unauthorized users have access to the axis interface, then the password should be configurable without a default in the module (people can still add a default in hiera, in our case in the common.yaml file). This is the reason the notification to the exec is a poor solution: if the password is changed in the puppet config, it is not updated. So it's better not to have `refreshonly => true`, but instead to use `onlyif` or `unless` with something `grep`. This is a bit tricky (one will need to escape regex metacharacters). For maximum robustness, this could of course be done in ruby.